analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/ProcessTestChecks.cs

```diff
@@ -18,45 +18,79 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
+using Microsoft.CodeAnalysis;
 using SonarAnalyzer.SymbolicExecution.Roslyn;
 using ProcessFunc = System.Func<SonarAnalyzer.SymbolicExecution.Roslyn.SymbolicContext, SonarAnalyzer.SymbolicExecution.Roslyn.ProgramState[]>;
 using ProcessFuncSimple = System.Func<SonarAnalyzer.SymbolicExecution.Roslyn.SymbolicContext, SonarAnalyzer.SymbolicExecution.Roslyn.ProgramState>;
 
 namespace SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution
```

Comment:
There could be one class, with a bool parameter in the constructor that would dictate if the PreProcess behavior should be modified or the PostProcess one. That way we would get rid of all the constructor duplications between the two classes.

Reply:
That would make usage of these more difficult. It is handy to have `new PreProcessTestCheck(...)` or `new PostProcessTestCheck(...)`.

In the best-case scenario. we will remove the `PreProcess` event completely from the engine. But that can be done only after we migrate all the rules.